src/services/operate.js

```diff
@@ -0,0 +1,22 @@
+import Big from 'big.js';
+
+export default function operate(numberOne, numberTwo, operation) {
+  try {
+    switch (operation) {
+      case '+':
+        return Big(numberOne).plus(numberTwo).toString();
+      case '-':
+        return Big(numberOne).minus(numberTwo).toString();
+      case 'x':
+        return Big(numberOne).times(numberTwo).toString();
+      case '÷':
+        return Big(numberOne).div(numberTwo).toString();
+      case '%':
+        return Big(numberOne).mod(numberTwo).toString();
+      default:
+        return 0;
+    }
+  } catch(e) {
+    alert('Something went wrong!');
```

Comment:
It's good that we are catching the division by zero error here. But since we're using React let us use the "react" way of handling errors. 

We can include a state in `App` for `error` (w/ default of `null`) and then in this catch statement we can make sure that that `state` is changed whenever the application encounters a division by zero error.

Reply:
Okay, I handled it covering the calculation function within a try-catch block and return an error state. It's still a declarative solution. I read some articles about how to handle errors in React but given solutions are all about handling errors in component scope. But in our scenario, we have the external calculation module and it looks it's the best way with handling it declaratively. Also, you may know it but official React docs recommend handling errors in an imperative way. Check it out if you want to see how:  https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html